Update okta-sdk-golang to v2.12.1

monde opened the pull request:
Updates okta-sdk-golang to v2.12.1
`go get github.com/okta/okta-sdk-golang/v2`
Removing any duplicate code in sdk/ package.

monde committed:
Update okta-sdk-golang to v2.11.1:
`go get github.com/okta/okta-sdk-golang/v2@v2.11.1`

monde committed:
Adjust code for okta-sdk-golang v2.11.1 changes.

monde committed:
Making use of okta-sdk-golang's Application.UploadApplicationLogo

monde committed:
Fixed TestAccAppOAuthApplication_postLogoutRedirectCrud . Doesn't seem like it ever tested correctly given the code changes?

monde committed:
Switching from local sdk authenticator to okta-sdk-golang Authenticator

monde committed:
Switching from local sdk profile mappings to okta-sdk-golang ProfileMapping

monde committed:
Fixes integration test TestAccOktaDataSourceIdpSocial_read

monde committed:
Fixes integration test TestAccOktaIdpSocial_crud

monde committed:
okta-sdk-golang has a fix for LinkedObjects ResponseLinks _links property.

monde committed:
Switching from local sdk linked objects to okta-sdk-golang LinkedObject

monde committed:
Removing as much custom API calls out of sdk/user_factor.go to pure okta-sdk-golang as possible.

monde committed:
Updated okta-sdk-golang having correct EnrollFactor

monde committed:
Updated okta-sdk-golang with corrected codegen for GetPolicy/UpdatePolicy

monde committed:
Replace local sdk policy code with okta-sdk-golang policy code where
equivalence exists. Make ACC test TestAccOktaPolicySignOn_crud passing.

monde committed:
update okta-sdk-golang to v2.12.1

monde committed:
Fix acceptance test TestAccOktaAppSignOnPolicyRule

monde committed:
Note how FF `SHARED_APP_SIGN_ON_POLICIES` affects resource `policy_profile_enrollment_apps`

monde committed:
Replace local sdk.Subscription with okta-sdk-golang Subscription

monde committed:
Making acceptance test TestAccOktaDefaultPasswordPolicy pass.

monde committed:
Use back off on the deactivate/delete application operations.

monde committed:
Fix TestAccOktaPolicyPassword_crud integration test.

monde committed:
Prep for v3.25.0

vijetmahabaleshwar-okta commented on website/docs/r/policy_profile_enrollment_apps.html.markdown:
```diff
@@ -8,7 +8,7 @@ description: |-
 # okta_policy_profile_enrollment_apps
 
-~> **WARNING:** This feature is only available as a part of the Identity Engine. [Contact support](mailto:dev-inquiries@okta.com) for further information.
+~> **WARNING:** This feature does not work with orgs having the feature flag `SHARED_APP_SIGN_ON_POLICIES` which may include orgs not fully upgraded to Identity Engine. [Contact support](mailto:dev-inquiries@okta.com) for further information.
```
Are you sure it's for orgs that have the FF `SHARED_APP_SIGN_ON_POLICIES`? or is it org that do NOT have `SHARED_APP_SIGN_ON_POLICIES`?

monde replied:
I'll undo that text. I'm not 100% on the claim now that I double checked it.

jphuynh committed:
docs/r/okta_app_saml: Minor updates to documentation for r/okta_app_saml

I was originally looking to fix the Optional part for `acs_endpoints` but then did an overall pass for the resource docs. Making wording consistent for default values and added some details about required arguments when `preconfigured_app` isn't provided.

monde committed:
Revert note about policy_profile_enrollment_apps

monde committed:
Merge branch 'master' into update_okta_sdk_golang

monde committed:
Re-do merge conflicts "fix"

monde committed:
Bump version on user-agent string for client.

monde committed:
Merge pull request #1076 from jphuynh/docs-r-okta_app_saml

docs/r/okta_app_saml: Minor updates to documentation for r/okta_app_saml

monde committed:
Note about @jphuynh's document update.
